spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -560,11 +560,13 @@ public Scan buildChangelogScan() {
     }
 
     boolean emptyScan = false;
+
     if (startTimestamp != null) {
+      boolean isBeforeCurrentSnapshot = table.currentSnapshot().timestampMillis() < startTimestamp;
```

Comment:
Should we check if table.currentSnapshot() is null? It is an empty scan if a table doesn't have any snapshot. I'm concerned about a NPE here.